### PR TITLE
Dependency check documentation

### PR DESCRIPTION
**Pull request by danielpetisme**
As discussed https://github.com/jhipster/generator-jhipster/issues/8191

### CODE REVIEW

**Comment by danielpetisme** [comment]
For the title, I used dependency check because it's the name of the OWASP plugins. If its misleading, I can rename it to **Dependency Vulnerabilities Check** (as https://github.com/jhipster/generator-jhipster/issues/6329).

The *WIP* in the client side section is because it I didn't find reliable tools (some of the one I used in the past have been acquired and not free anymore or not maintained).
Because I can be a little bit more verbose and precise than we need help for this section to encourage contributions.

Otherwise, for me, it's done.

File: pages/dependency-check.md [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+  ..
+  </plugins>
+  ...
+</build>
+```
+You can now run `./mvnw verify` to generate a dependency check report under `target`.
+
+### Gradle
+[OWASP Gradle Dependency Check plugin documentation](https://jeremylong.github.io/DependencyCheck/dependency-check-gradle/index.html)
+Create a file `gradle/security.groovy` with the following content
```
Comment (atomfrede) [inline pages/dependency-check.md]: it should be `gradle/security.gradle` (below it is correct)

File: pages/dependency-check.md [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+        classpath 'org.owasp:dependency-check-gradle:3.3.1'
+    }
+}
+
+apply plugin: 'org.owasp.dependencycheck'
+if(project.hasProperty('strict-security')) {
+  check.dependsOn dependencyCheckAnalyze
+}
+
+Add `apply from: 'gradle/security.gradle'` in your `build.groovy` script.
```
Comment (atomfrede) [inline pages/dependency-check.md]: should be `build.gradle` :)

File: pages/dependency-check.md [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+  </plugins>
+  ...
+</build>
+```
+You can now run `./mvnw verify` to generate a dependency check report under `target`.
+
+### Gradle
+[OWASP Gradle Dependency Check plugin documentation](https://jeremylong.github.io/DependencyCheck/dependency-check-gradle/index.html)
+Create a file `gradle/security.groovy` with the following content
+```
```
Comment (atomfrede) [inline pages/dependency-check.md]: Can't we get rid of the separate script file and add it to the main build file? As we need to modify `build.gradle` anyways we can add it to the main file imho.

**Comment by jdubois** [comment]
`Dependency Vulnerabilities Check` looks good to me. I'll probably also reword a few sentences.
(sorry I'm late to review the PR at the moment)

**Comment by danielpetisme** [comment]
Renaming done.
Do you want me to rephrase some parts? Otherwise, sounds my job is done 😉

**Comment by deepu105** [comment]
Btw NPM now automatically checks this when triggering install

**Comment by danielpetisme** [comment]
@deepu105 great. Do you have any doc I could use?

**Comment by deepu105** [comment]
https://docs.npmjs.com/getting-started/running-a-security-audit

It even has a handy audit fix command

**Comment by danielpetisme** [comment]
Done, I didn't go into details because the NPM page is awesome and complete.

Do you know if somewhere in the community someone made a "JHipster + Security = ❤️ " talk? It could be an interesting topic. Go trough OWASP top ten and present how JHipster mitigate them.

Maybe next DevoxxFR 😉

**Comment by jdubois** [comment]
@danielpetisme looks like an awesome idea for a talk! I know several companies pay big $$$ to have security audits of their JHipster application, but they never give us an feedback (or any contribution). So that would hopefully help people to test & contribute on this part, which is of course very important.

I'll review your page and probably change some wording.

**Pull request merged**